jib-cli/src/integration-test/java/com/google/cloud/tools/jib/cli/cli2/JarCommandTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.cli2;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.tools.jib.Command;
+import com.google.common.io.Resources;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import picocli.CommandLine;
+
+public class JarCommandTest {
+
+  private static final Integer SUCCESS_CODE = 0;
+  private static final Integer FAILURE_CODE = 1;
+
+  private final PrintStream originalOut = System.out;
+  private final PrintStream originalErr = System.err;
+  private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+  private final ByteArrayOutputStream err = new ByteArrayOutputStream();
+
+  @Before
+  public void setUp() {
+    out.reset();
+    err.reset();
+    System.setOut(new PrintStream(out));
```

Comment:
This is a creative trick, but I feel it isn't really safe to replace stdout and stderr. For example, the unit framework depends on it, and I think we will not be able to see test failures.

I have been thinking of two options for verifying command output: run as a separate process, or refactor the code. I didn't really like refactoring code only to be able to check log messages. Eventually I think we will reach the point where we really need to check log messages. And we will eventually find a way to run Jib CLI as a standalone process. Maybe for now, if we can't easily find a way to check log messages, I'm fine punting it.

Reply:
Ah yeah this doesn't seem too great. Maybe we should create an issue to make the logger use `PrintWriter` instead of printing to `System.err` or `System.out` - https://picocli.info/#_testing_your_application.

Reply:
Oh, nice documentation. I think `setOut` and `setErr` will just work without us having to change the logger. Have you tried that?

Reply:
I agree haha. And sorry, are you referring to this approach?:
```
MyApp app = new MyApp();
CommandLine cmd = new CommandLine(app);
StringWriter sw = new StringWriter();
cmd.setOut(new PrintWriter(sw));
```

Reply:
Yes. I feel like this will work.

Reply:
When I had attempted this approach the output and the error messages could not be set using `setOut/Err` and were getting printed onto the console. I think it's because we are currently directly printing to `System.out/err` in the logger instead of writing to the picocli writer? `This assumes that the application uses the PrintWriter provided by CommandLine.getOut or CommandLine.getErr`

Reply:
Ah, you're right. Let's change the logger. I think it'll be pretty straightforward.

Reply:
Sounds good! Added #2886.

Reply:
:+1: Let's fix #2886 first before this PR.

Reply:
Done.